tests/HashMapProperties.hs

```diff
@@ -150,6 +150,10 @@ pDeleteCollision k1 k2 k3 idx = (k1 /= k2) && (k2 /= k3) && (k1 /= k3) ==>
 pInsertWith :: Key -> [(Key, Int)] -> Bool
 pInsertWith k = M.insertWith (+) k 1 `eq_` HM.insertWith (+) k 1
 
+pInsertWithKey :: Key -> [(Key, Int)] -> Bool
+pInsertWithKey k = M.insertWithKey f k 1 `eq_` HM.insertWithKey f k 1
+  where f (K k') a b = if k' >= 0 then a else b
+
```

Comment:
Neither this nor the pre-existing `pInsertWith` is particularly nice. They both need a lot of thinking to decide if they test enough, and `pInsertWith` definitely does *not*. I think we want something like

```haskell
pInsertWithKey :: Fun (Key, A, A) A -> Key -> A -> [(Key, A)] -> Bool
```

Reply:
Updated both of them to use `Fun`.